src/createEventEmitterProxy.ts

```diff
@@ -93,8 +121,33 @@ export function createEventEmitterProxy<Type extends EventEmitterLike>(
       }
 
       const value = target[name];
-      if (value instanceof Function) {
+
+      if (typeof value === 'function') {
         return function (this: unknown, ...args: any[]) {
+          if (name === 'once') {
+            const unwrappedHandler = args[1];
+            const wrappedHandler = (...handlerArgs: any[]) => {
+              removeEvent(args[0], wrappedHandler);
+              return unwrappedHandler(...handlerArgs);
+            };
+            args[1] = wrappedHandler;
+          }
+          if (
+            name === 'on' ||
+            name === 'addListener' ||
+            name === 'prependListener' ||
+            name === 'once'
+          ) {
+            eventsAdded.push({
+              addedWith: name,
+              name: args[0],
+              handler: args[1],
```

Comment:
For `once`, should we be storing the unwrapped handler here? If someone tries to remove that listener, they'd be passing in the unwrapped handler, not the wrapped one. So the comparison in `removeEvent` would always fail to remove it.

Reply:
It would be great to add a test case for this as well (i.e. add an event listener using `once`, try to remove it, then confirm that it has been removed and hasn't been migrated)

Reply:
ah yes nice catch!

Reply:
Okay I've updated PR accordingly. Lmk what you think

Reply:
(this test should be what you were highlighting https://github.com/MetaMask/swappable-obj-proxy/pull/53/commits/578a5a0a7ffc3a8ad3f7da23c89564511e67d8ab#diff-5959fa448c7362c6ed2d30b45343e71a7a4e13358ea4fde75e9871e0b3fdad2eR271)

Reply:
Nice! Looks like it should work now, and the test coverage looks good